src/Search/index.js

```diff
@@ -0,0 +1,125 @@
+/**
+ *  Copyright 2020 Adobe. All rights reserved.
+ *  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License. You may obtain a copy
+ *  of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under
+ *  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *  OF ANY KIND, either express or implied. See the License for the specific language
+ *  governing permissions and limitations under the License.
+ */
+
+/** @jsx jsx */
+import { css, jsx } from '@emotion/core'
+import { useState } from 'react'
+import PropTypes from 'prop-types'
+import { navigate, Link } from 'gatsby'
+import { Index } from 'elasticlunr'
+import { Heading, Text } from '@react-spectrum/text'
+import { Item, Menu } from '../Menu'
+import { Popover } from '../Popover'
+import { SearchField } from '../SearchField'
+
+import './search.css'
+
+const Search = ({ searchIndex = {}, ...props }) => {
+  const [index] = useState(Index.load(searchIndex))
+  const [results, setResults] = useState([])
+  const [items, setItems] = useState([])
+  const [isOpen, setIsOpen] = useState(false)
+
+  const search = (searchTerm) => {
+    const searchResults = index
+      .search(searchTerm, { expand: true })
+      .map(({ ref }) => {
+        return index.documentStore.getDoc(ref)
+      })
+
+    setResults(searchResults)
+    const docsResultMenuItems = [
+      <Item key='docs-divider'>
+        <h5>Docs</h5>
+      </Item>
+    ]
+    const apiResultMenuItems = [
+      <Item key='api-divider'>
+        <h5>APIs</h5>
+      </Item>
+    ]
+    for (const result of searchResults) {
+      const item = (
+        <Link key={result.id} className='searchMenuLink' to={result.path}>
+          <Item>{result.title}</Item>
+        </Link>
+      )
+      if (result.type === 'apis') {
+        apiResultMenuItems.push(item)
+      } else {
+        docsResultMenuItems.push(item)
+      }
+    }
+    const topResultMenuItems = []
+    if (docsResultMenuItems.length > 1)
+      topResultMenuItems.push(...docsResultMenuItems)
+    if (apiResultMenuItems.length > 1)
+      topResultMenuItems.push(...apiResultMenuItems)
```

Comment:
we can probably move this logic out of `search`.
every time `setResults()` is called, the component will re-render with a new value for `results`.
we can then run this logic and save the result to `items` to use later in the resulting jsx.

Reply:
Refactored it out of the `search` function.